opba-banking-rest-impl/src/main/java/de/adorsys/opba/tppbankingapi/service/BankService.java

```diff
@@ -23,15 +23,17 @@
     private final BankSearchRepositoryImpl bankRepository;
 
     @Transactional(readOnly = true)
-    public Optional<BankProfileDescriptor> getBankProfile(UUID profileUuid) {
+    public Optional<BankProfileDescriptor> getBankProfile(UUID profileUuid, boolean onlyActive) {
         return bankProfileJpaRepository.findByUuid(profileUuid)
+                .filter(profile -> !onlyActive || profile.isActive())
                 .map(BankProfile.TO_BANK_PROFILE_DESCRIPTOR::map);
     }
 
     @Transactional(readOnly = true)
-    public List<BankDescriptor> getBanks(String query, int startPos, int maxResults) {
+    public List<BankDescriptor> getBanks(String query, int startPos, int maxResults, boolean onlyActive) {
         return bankRepository.getBanks(query, startPos, maxResults)
                 .stream()
+                .filter(bank -> !onlyActive || bank.isActive())
```

Comment:
Use filtering directly in the query, as this kind of filtering breaks pagination (I.e. page that has all-inactive banks will yield empty bank response)

Reply:
filtering was moved to query